File: Lab4/main/java/window/controller/Receiver.java
package window.controller;

import window.model.Color;
import window.model.MyInterface;
import window.model.WindowProxy;
import window.model.WindowService;

public class Receiver implements MyInterface {
    private WindowService win;

    public Receiver(WindowService win) {
        this.win = win;
    }

    @Override
    public void openWindow() {
        win.openWindow();
        System.out.println(win);
    }

    @Override
    public void wordsCounter(String userText) {
        win.wordsCounter(userText);
    }

    @Override
    public void setWindowColor(Color color) {
        MyInterface window = (MyInterface) WindowProxy.newProxyInstance(win);
        try {
            window.setWindowColor(Color.Violet);
        } catch (Exception e) {
            e.printStackTrace();
        }
    }

    @Override
    public void closeWindow() {
        win.closeWindow();
        System.exit(0);
    }
}

File: Lab6/src/main/kotlin/controller/Receiver.kt
package controller

import model.MyInterface
import model.WindowService
import kotlin.system.exitProcess

class Receiver(val win: WindowService): MyInterface {
    override fun closeWindow() {
        win.closeWindow()
        exitProcess(0)
    }

    override fun openWindow() {
        win.openWindow()
        println(win)
    }

    override fun wordsCounter(userText: String) {
        win.wordsCounter(userText)
    }
}